app/models/user.rb

```diff
@@ -175,6 +175,7 @@ def course_roles(course)
   def highest_role(course)
     roles = course_roles(course)
 
+    return CoursesUsers::Roles::INSTRUCTOR_ROLE if admin?
```

Comment:
This should be done before the query for roles, as it may replace the need to make a database query.